src/main/java/com/pretz/dyntreeserver/UserDao.java

```diff
@@ -1,9 +1,12 @@
 package com.pretz.dyntreeserver;
 
 import com.pretz.dyntreeserver.domain.User;
-import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface UserDao extends CrudRepository<User, Long> {
+import java.util.Optional;
 
-    public User findByEmail(String email);
+public interface UserDao extends JpaRepository<User, Long> {
+
+    Optional<User> findByNameAndPassword(String name, String password);
```

Comment:
What's the idea of `findByNameAndPassword`? isn't `findByName` enough?

Reply:
It probably is. This part of code isn't set in stone yet, as you maybe remember we have talked about using only email as login or even integrating Facebook login service in the app.